src/compiler/parser.ts

```diff
@@ -2472,6 +2482,9 @@ namespace ts {
                     return parseTupleType();
                 case SyntaxKind.OpenParenToken:
                     return parseParenthesizedType();
+                case SyntaxKind.PartialKeyword:
+                    const partialNode = tryParse(parsePartialType);

```

Comment:
Ah, it looks like my question above is answered here - that's what it looks like is happening @sandersn.

This is going to have some weird behavior because a lot of the time people will just get a very confusing error message due to the way function bodies will be parsed.

```ts
interface partial { /*...*/ }
function f(x: number): partial {

}
// error: expected function body
```

Why not just make `partial` a reserved type keyword and make the error more obvious?